cmstestsuite/unit_tests/grading/__init__.py

```diff
@@ -77,74 +90,94 @@ def _res(execution_time, execution_wall_clock_time, execution_memory,
             r["filename"] = filename
         return r
 
+    def assertRes(self, r0, r1):
+        """Assert that r0 and r1 are the same result."""
+        self.assertAlmostEqual(r0["execution_time"], r1["execution_time"])
```

Comment:
Would you mind adding something equivalent to `assert {"execution_time", "execution_wall_clock_time", "execution_memory", "syscall", "signal", "filename"}.issubset(iterkeys(r0))` and similarly for `r1`?

Reply:
Apart that it'd be wrong (the last three are usually not present), I disagree. I might add set(iterkeys(r0)) == set(iterkeys(r1)), if that's your goal?

Reply:
Agreed offline not to add anything.